Components/src/iosMain/kotlin/dispatcher.kt

```diff
@@ -0,0 +1,43 @@
+package com.splendo.kaluga
+
+import kotlinx.coroutines.*
+import platform.darwin.*
+import kotlin.coroutines.CoroutineContext
+
+/*
+
+Copyright 2019 Splendo Consulting B.V. The Netherlands
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+@UseExperimental(InternalCoroutinesApi::class)
+internal class NsQueueDispatcher(private val dispatchQueue: dispatch_queue_t) : CoroutineDispatcher(), Delay {
```

Comment:
I also thought that. I tried `MainScope().launch` and `GlobalScope.launch(Dispatchers.Main)` and expected to run them in main queue on iOS. But in all cases I've got an exception:

`kotlin.IllegalStateException: There is no event loop. Use runBlocking { ... } to start one.`

Looks like this one https://github.com/Kotlin/kotlinx.coroutines/issues/470 is still open and not included into library, and we have to keep this workaround. Or I missed something...

Reply:
Look the `Components` examples, you need to wrap the launch in `runBlocking` for now.

Reply:
Perhaps we should make a convenience `launch` method, because despite of the error message people are confused by this.

Reply:
Still not clear how to use it then. We have `suspend alert` to build alert and `suspend show` to show. If we wrap all with `runBlocking` it will never return because show is waiting for user's input.

Reply:
You wrap the `launch` inside `runBlocking`.  `runBlocking` will not wait for what happens inside `launch`.

Did you check the example of Components?

Reply:
You mean example of LocationPrinter?

Reply:
I'd look at all of them, also to see how common testing is done (#56), but yeah just search `launch` or `MainScope` and you can see the usage.

Reply:
So, in `AlertFactory` in shared example we have three functions.
Last one works with `runBlocking { MainScope().launch { ... }}` fine.
But this approach doesn't work if I have `delay` call inside with the same event loop complains. And this doesn't work for `CancelableContinuation` (when we call `resume`) for the same reasons.